### DISCUSSION TITLE
Publish in server

### DISCUSSION THREAD

**Discussion by iranandroid** [filamentphp/filament#3841] [Help] [answered]
Hi eveyone.
I changed Login.php class in vendor folder then publish config and work fine in locally. but when push code in git and publish in server with CI/DI, filament download automatically
how can publish filament all changes?

**Comment by danharrin** [comment, marked as answer]
You should never change files in your /vendor folder.

In this case, create a new Login class that contains your changes, and then use that class namespace inside the config file, to change the login page.

**Reply by iranandroid** [reply]
Hi @danharrin 
Thank you for replay. I create new class with name ```Login.php``` in ```\App\Http\Controllers\Auth``` folder and change config file to this:
```
    'auth' => [
        'guard' => env('FILAMENT_AUTH_GUARD', 'web'),
        'pages' => [
            'login' => \App\Http\Controllers\Auth\Login::class
        ],
    ],
```
But when open page and login in panel, I get the following error:
![image](https://user-images.githubusercontent.com/20900080/188297070-283d0f63-68f0-4b2e-8485-d73c2451c072.png)

**Reply by iranandroid** [reply]
@danharrin  I Solved my problem. Thank you.
